lib/Browser.js

```diff
@@ -31,6 +31,50 @@ class Browser extends EventEmitter {
     this._screenshotTaskQueue = new TaskQueue();
     this._connection = connection;
     this._closeCallback = closeCallback || new Function();
+    /** @type {Map<string, Target>} */
+    this._targets = new Map();
+    this._connection.on('Target.targetCreated', this._targetCreated.bind(this));
+    this._connection.on('Target.targetDestroyed', this._targetDestroyed.bind(this));
+  }
+
+  /**
+   * @param {!Connection} connection
+   * @param {boolean} ignoreHTTPSErrors
+   * @param {function()=} closeCallback
+   */
+  static async create(connection, ignoreHTTPSErrors, closeCallback) {
+    const browser =  new Browser(connection, ignoreHTTPSErrors, closeCallback);
+    await connection.send('Target.setDiscoverTargets', {discover: true});
+    return browser;
+  }
+
+  /**
+   * @param {{targetInfo: {targetId: string, type:string}}} param0
+   */
+  async _targetCreated({ targetInfo }) {
+    const target = await this._targetForInfo(targetInfo);
+    this.emit(Browser.Events.TargetCreated, target);
+  }
+
+  /**
+   * @param {{targetId: string}} param0
+   */
+  async _targetDestroyed({targetId}) {
+    const target = this._targets.get(targetId);
+    if (!target)
+      return;
+    this._targets.delete(targetId);
+    this.emit(Browser.Events.TargetDestroyed, target);
+  }
+
+  /**
+   * @param {{targetId: string, type: string}} targetInfo
+   * @return {Target}
+   */
+  _targetForInfo(targetInfo) {
```

Comment:
nit: this kind of methods are usually prefixed with `_ensure`: `_ensureTargetForInfo`